docs/deployment/openshift.md

```diff
@@ -1,4 +1,286 @@
 # Deployment guide for OpenShift
 
-Coming soon.
+OpenFaaS is Kubernetes-native and uses *Deployments*, *Services* and *Secrets*. For more detail check out the ["faas-netes" repository](https://github.com/openfaas/faas-netes).
 
+Use this guide to deploy OpenFaaS to an OpenShift distribution running an OpenShift version between 3.9 and 4.0.
+
+## Build a cluster
+
+You can start evaluating FaaS and building functions on your laptop or on a VM (cloud or on-prem).
+
+* [Install guide for minishift](https://docs.okd.io/latest/minishift/getting-started/installing.html)
+
+* [Deployment information on running OpenShift ](https://www.openshift.com/products)
+
+### Install the `faas-cli`
+
+You can install the OpenFaaS CLI using `brew` or a `curl` script.
+
+* via `brew`:
+
+```bash
+brew install faas-cli
+```
+
+* via `curl`:
+
+```bash
+$ curl -sL https://cli.openfaas.com | sudo sh
+```
+
+If you run the script as a normal non-root user then the script will be downloaded to the current folder.
+
+### Pick Minishift or OpenShift Installation (A or B)
+
+OpenShift is slightly different than the Helm installation for the vanilla kubernetes installation, we will need to apply the yaml files recursively. 
+
+#### A. Deploy for Minishift
+
+A minishift addon is avaiable [here](https://github.com/mirroredge/openfaas-minishift-addon)
+
+Once cloned navigate inside the repo run:
+``` bash
+$ minishift addons install openfaas
+```
+
+Then to apply it to the running minishift instance:
+``` bash
+$ minishift addons apply openfaas
+```
+
+To enable it every restart of your minishift instance:
+``` bash
+$ minishift addons enable openfaas
+```
+
+#### B. Deploy for OpenShift (for production)
+
+This step assumes you have a running OpenShift cluster and have the oc client installed
+
+* Deploy the whole stack
+
+    This command is split into two parts so that the OpenFaaS namespaces are always created first:
+
+    * openfaas - for OpenFaaS services
+    * openfaas-fn - for functions
+
+    ```bash
+    $ oc new-project openfaas
+    $ oc new-project openfaas-fn
+    ```
+
+    Now deploy OpenFaaS:
+
+    ```bash
+    $ cd faas-netes && \
+    oc apply -R -f ./openshift
```

Comment:
Where does this folder come from? Is there a PR in faas-netes to add this?  I looked but perhaps I missed it.

Reply:
Correct. Its in this PR: https://github.com/openfaas/faas-netes/pull/407